resolver/resolver_test.go

```diff
@@ -351,6 +351,38 @@ func TestHTTP(t *testing.T) {
 	}
 }
 
+// TestHTTPAcceptApplicationJson tests that valid requests that specify
+// 'Accept: application/json' succeed. This used to fail with
+// 406 Not Acceptable.
+// See https://jira.mesosphere.com/browse/DCOS_OSS-611
+func TestHTTPAcceptApplicationJson(t *testing.T) {
+	// setup DNS server (just http)
+	res, err := fakeDNS()
+	if err != nil {
+		t.Fatal(err)
+	}
+	res.version = "0.1.1"
```

Comment:
I had a look. The `res.version` is set in `TestHTTP` because we test the `/v1/version` endpoint in that test. In this new test it is unnecessary and I have removed it. I've squashed the commits and will merge this once the the checks all pass. Thanks for picking this up @jdef